c2rust-transpile/src/translator/structs.rs

```diff
@@ -46,19 +46,17 @@ fn contains_block(expr_kind: &Expr) -> bool {
     match expr_kind {
         Expr::Block(..) => true,
         Expr::Assign(ExprAssign { left, right, .. }) => {
-            contains_block(&left) || contains_block(&right)
+            contains_block(left) || contains_block(right)
         }
         Expr::AssignOp(ExprAssignOp { left, right, .. }) => {
-            contains_block(&left) || contains_block(&right)
+            contains_block(left) || contains_block(right)
         }
         Expr::Binary(ExprBinary { left, right, .. }) => {
-            contains_block(&left) || contains_block(&right)
+            contains_block(left) || contains_block(right)
         }
-        Expr::Unary(ExprUnary { expr, .. }) => contains_block(&expr),
-        Expr::MethodCall(ExprMethodCall { args, .. }) => {
-            args.iter().map(|e| contains_block(&e)).any(|b| b)
-        }
-        Expr::Cast(ExprCast { expr, .. }) => contains_block(&expr),
+        Expr::Unary(ExprUnary { expr, .. }) => contains_block(expr),
+        Expr::MethodCall(ExprMethodCall { args, .. }) => args.iter().any(contains_block),
+        Expr::Cast(ExprCast { expr, .. }) => contains_block(expr),
         _ => false,
     }
```

Comment:
Can the `contains_block` be refactored out of the `match` by returning an `Iterator` and then calling `.iter().any(contains_block)` on it outside of the `match`?